### PR TITLE
Remove dead gallery code 

### PR DESCRIPTION
**Pull request by Erin007**
As part of long-overdue clean up I'm removing GalleryActivities and associated dead code. 

### studio.code.org/gallery 
Years ago we made a project gallery for Artist and PlayLab projects at studio.code.org/gallery:

Screen Shot 2020-03-12 at 8 36 53 PM

We no longer support this page in favor of studio.code.org/projects. 

### homepage 

 There was a mini, personalized version of what was shown at studio.code.org/gallery on the home page when it was on Pegasus. When we redesigned the home page, re-built it in React and moved it to Dashboard, we replaced that version of a gallery with this:
Screen Shot 2020-03-12 at 8 05 34 PM
The current projects section of the homepage pulls projects from storage apps rather than gallery activities.

### CODE REVIEW

File: dashboard/db/schema.rb [hunk 1]
```diff
@@ -1513,6 +1513,8 @@
     t.integer  "score"
     t.datetime "created_at",    null: false
     t.datetime "updated_at",    null: false
+    t.index ["created_at"], name: "index_teacher_scores_on_created_at", using: :btree
+    t.index ["teacher_id"], name: "index_teacher_scores_on_teacher_id", using: :btree
```
Comment (breville) [inline dashboard/db/schema.rb]: Are these changes intended?

File: dashboard/app/controllers/home_controller.rb [hunk 2]
```diff
@@ -68,14 +65,6 @@ def home
     render 'home/index'
   end
 
-  def gallery_activities
```
Comment (davidsbailey) [inline dashboard/app/controllers/home_controller.rb:71]: please also remove the corresponding config from routes.rb: https://github.com/code-dot-org/code-dot-org/blob/45746b927254f76e682440febcf9bf09f7db674f/dashboard/config/routes.rb#L30-L35

**Review by davidsbailey — COMMENTED** [review]
Thank you for tackling this, @Erin007 ! Just want to make sure we are doing due diligence here so I checked Google Analytics which says that some people are still using this page: https://analytics.google.com/analytics/web/#/report/content-pages/a37745279w66217109p68074336/explorer-table.filter=~2Fgallery&explorer-table.plotKeys=%5B%5D/

This lead me to look around for links to this page.  here are a few:
1. https://code.org/report-abuse --> https://github.com/code-dot-org/code-dot-org/blob/f17049999651c0e01ca0fbb0a8807ea7e73c0e2a/pegasus/sites.v3/code.org/public/report-abuse.md.erb#L8
2.  this one appears to be unused and can be safely removed: https://github.com/code-dot-org/code-dot-org/blob/9c50ccb77cf5af66dd092716eb007cd9f0eeab92/dashboard/app/helpers/course_block_helper.rb#L40-L42

At a minimum, (1) needs to be removed or replaced with a link to our public projects gallery, so that we are no longer pointing people to an invalid url.

Additionally, I think it would be good to start redirecting from /gallery to the public projects gallery, given that traffic has not quite dropped to zero yet.

As a follow-up step, which I want to document here but which you shouldn't feel obligated to do, we should also stop adding things to this table. Some code which handles that can be found by searching for `save_to_gallery_url`.

**Comment by Erin007** [comment]
> Thank you for tackling this, @Erin007 ! Just want to make sure we are doing due diligence here so I checked Google Analytics which says that some people are still using this page: https://analytics.google.com/analytics/web/#/report/content-pages/a37745279w66217109p68074336/explorer-table.filter=~2Fgallery&explorer-table.plotKeys=%5B%5D/
> 
> This lead me to look around for links to this page. here are a few:
> 
> 1. https://code.org/report-abuse --> https://github.com/code-dot-org/code-dot-org/blob/f17049999651c0e01ca0fbb0a8807ea7e73c0e2a/pegasus/sites.v3/code.org/public/report-abuse.md.erb#L8
> 2. this one appears to be unused and can be safely removed: https://github.com/code-dot-org/code-dot-org/blob/9c50ccb77cf5af66dd092716eb007cd9f0eeab92/dashboard/app/helpers/course_block_helper.rb#L40-L42
> 
> At a minimum, (1) needs to be removed or replaced with a link to our public projects gallery, so that we are no longer pointing people to an invalid url.
> 
> Additionally, I think it would be good to start redirecting from /gallery to the public projects gallery, given that traffic has not quite dropped to zero yet.
> 
> As a follow-up step, which I want to document here but which you shouldn't feel obligated to do, we should also stop adding things to this table. Some code which handles that can be found by searching for `save_to_gallery_url`.

All great points, Dave! I'd started on some of them here: https://github.com/code-dot-org/code-dot-org/pull/33618 but rather than splitting up, maybe this is a good task to tackle as a mega PR so we know we got out all of the tendrils at the same time.  I'll keep working on it, and let you know when I'm ready for another look.  Thanks!

**Review by wjordan — APPROVED** [review]
I don't have much else to add to the detailed comments already made, but in general this change sounds good to me!

**Comment by Erin007** [comment]
OK @davidsbailey This is ready for another look. Anything you can think of that I missed? Is this consistent with what you were [suggesting about impressive levels](https://codedotorg.slack.com/archives/C1B3PNDL7/p1585605941015400)? Thanks!

File: dashboard/config/routes.rb [hunk 3]
```diff
@@ -187,6 +181,8 @@
     end
   end
 
+  get "/gallery", to: redirect("/projects")
```
Comment (davidsbailey) [inline dashboard/config/routes.rb]: nit: I think this should redirect to /projects/public, since /gallery was the old public gallery (right?) and /projects/public is the new public gallery

File: apps/src/p5lab/P5Lab.js [hunk 4]
```diff
@@ -1495,11 +1495,6 @@ P5Lab.prototype.displayFeedback_ = function() {
     // add 'impressive':true to non-freeplay levels that we deem are relatively impressive (see #66990480)
     showingSharing:
       !level.disableSharing && level.freePlay /* || level.impressive */,
```
Comment (davidsbailey) [inline apps/src/p5lab/P5Lab.js]: comments regarding `impressive` can safely be removed from the comments on 2 lines here

File: dashboard/config/locales/en.yml [hunk 5]
```diff
@@ -22,10 +22,10 @@
 en:
   errors:
     messages:
-      not_saved: 'Please correct the following errors:'
-      too_young: 'This content has age restrictions in place and is not available for younger students.'
+      not_saved: "Please correct the following errors:"
+      too_young: "This content has age restrictions in place and is not available for younger students."
```
Comment (davidsbailey) [inline dashboard/config/locales/en.yml]: revert accidental changes to this file?

**Review by davidsbailey — CHANGES_REQUESTED** [review]
awesome work @Erin007, this is really coming along ! One thing which needs to be fixed:
* http://localhost.code.org:3000/report-abuse still shows a link to http://localhost.code.org:3000/gallery . This is the line which needs to be updated: https://github.com/code-dot-org/code-dot-org/blob/f17049999651c0e01ca0fbb0a8807ea7e73c0e2a/pegasus/sites.v3/code.org/public/report-abuse.md.erb#L8

and a few more loose ends which can be removed:
* https://github.com/code-dot-org/code-dot-org/blob/1cad5ac667f9ed1a3aa31063e6d06bd21a78e2bd/apps/src/eval/eval.js#L648-L649
* https://github.com/code-dot-org/code-dot-org/blob/2c83e67fa6c0c5813425e1edc64a995260683400/apps/src/feedback.js#L62
* `save_to_gallery` in reporting.js

future work (apologies if this is already captured somewhere):
* remove GalleryActivity model
* remove gallery_activties table (16M rows)

**Comment by Erin007** [comment]
Ok, @davidsbailey I think I got everything you requested this time. Anything else for this PR?

 The remainder of related tasks are tracked in [LP-1386](https://codedotorg.atlassian.net/browse/LP-1386)

Thank you!

**Review by davidsbailey — APPROVED** [review]
Looks great, @Erin007 ! Thanks for following through on this!
